tasks/src/resolver.rs

```diff
@@ -0,0 +1,87 @@
+use imap_flow::{client::ClientFlow, Flow, FlowInterrupt};
+use imap_types::response::{Response, Status};
+use tracing::warn;
+
+use crate::{Scheduler, SchedulerError, SchedulerEvent, Task, TaskHandle};
+
+/// The resolver is a scheduler than manages one task at a time.
+pub struct Resolver {
+    scheduler: Scheduler,
+}
+
+impl Flow for Resolver {
+    type Event = SchedulerEvent;
+    type Error = SchedulerError;
+
+    fn enqueue_input(&mut self, bytes: &[u8]) {
+        self.scheduler.enqueue_input(bytes);
+    }
+
+    fn progress(&mut self) -> Result<Self::Event, FlowInterrupt<Self::Error>> {
+        self.scheduler.progress()
+    }
+}
+
+impl Resolver {
+    /// Create a new resolver.
+    pub fn new(flow: ClientFlow) -> Self {
+        Self {
+            scheduler: Scheduler::new(flow),
+        }
+    }
+
+    /// Enqueue a [`Task`] for immediate resolution.
+    pub fn run_task<T: Task>(&mut self, task: T) -> TaskResolver<T> {
```

Comment:
I'm not quite happy with the current names `Resolver`, `Resolver::run_task` and `TaskResolver`.

Some alternatives:

- `Resolver`, `Resolver::resolve`, `ResolvingTask`
- `Runner`, `Runner::run`, `RunningTask`

But I'm just nitpicking. I you like we can keep the current names for now.

Reply:
I prefer resolve over run, because it gives this information of running a task till the end. So I opted for the first one :ok_hand: